### PR TITLE
rebase to MMTk-0.1

### PR DESCRIPTION
**Pull request by javadamiri**
This PR updates mmtk-v8 to work with MMTk-0.1.
It also includes some unused code clean-up.

### CODE REVIEW

File: mmtk/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ lto = true
 libc = "0.2"
 lazy_static = "1.1"
 log = "*"
-mmtk = { git = "ssh://git@github.com/mmtk/mmtk-core.git", rev = "d82284eaf2ae5e443eb10c236e24f616719bd831" }
+mmtk = { git = "ssh://git@github.com/mmtk/mmtk-core.git", rev = "0615b01837323221bf1d897c5c88bd25a80e3904" }
```
Comment (qinsoon) [inline mmtk/Cargo.toml:18]: You can point the rev of 0.1 `5cd2407424061db66cdb8573fac30704b948a11c`

**Review by qinsoon — APPROVED** [review]
LGTM. I will add the label to trigger CI. Can you check if the CI works? Last time when I tested it, the CI failed in a step to get v8 code.

**Comment by qinsoon** [comment]
Can you also change the trigger of the workflow to this?
```
on:
  pull_request:
    types: [opened, synchronize, reopened, labeled]
    branches:
      - master
```

Otherwise, labelling the PR will not trigger the workflow.

**Comment by javadamiri** [comment]
> LGTM. I will add the label to trigger CI. Can you check if the CI works? Last time when I tested it, the CI failed in a step to get v8 code.

It hasn't changed since its last successful run, but I'll check

**Comment by javadamiri** [comment]
> Can you also change the trigger of the workflow to this?
> 
> ```
> on:
>   pull_request:
>     types: [opened, synchronize, reopened, labeled]
>     branches:
>       - master
> ```
> 
> Otherwise, labelling the PR will not trigger the workflow.

done

**Pull request merged**